LEGO1/lego/legoomni/src/build/legocarbuild.cpp

```diff
@@ -771,14 +793,214 @@ MxS32 LegoCarBuild::FUN_10024850(MxLong p_x, MxLong p_y)
 	return result;
 }
 
-// STUB: LEGO1 0x10024890
-// STUB: BETA10 0x1006d512
-undefined4 LegoCarBuild::FUN_10024890(LegoEventNotificationParam* p_param)
+#ifdef NDEBUG
+
+// FUNCTION: LEGO1 0x10024890
+undefined4 LegoCarBuild::FUN_10024890(MxParam* p_param)
 {
-	// TODO
-	return 0;
+	undefined4 result = 0;
+	LegoControlManagerNotificationParam* param = (LegoControlManagerNotificationParam*) p_param;
+	assert(m_buildState);
+
+	if (param->m_unk0x28) {
+		switch (param->m_clickedObjectId) {
+		// The enum values are all identical between CopterScript, DunecarScript, JetskiScript, and RacecarScript
+		case CopterScript::c_Info_Ctl:
+			if (m_buildState->m_animationState != LegoVehicleBuildState::e_unknown4 &&
+				m_buildState->m_animationState != LegoVehicleBuildState::e_unknown2 &&
+				m_buildState->m_animationState != LegoVehicleBuildState::e_exiting &&
+				GameState()->GetCurrentAct() != LegoGameState::e_act2) {
+				if (m_unk0x108 > 0) {
+					DeleteObjects(&m_atomId, 500, 510);
+				}
+
+				m_unk0x258->SetUnknown0xbc(0);
+				m_destLocation = LegoGameState::e_infomain;
+				TransitionManager()->StartTransition(MxTransitionManager::e_mosaic, 50, FALSE, FALSE);
+				result = 1;
+			}
+
+			break;
+		case CopterScript::c_Exit_Ctl:
+			if (m_buildState->m_animationState != LegoVehicleBuildState::e_exiting &&
+				m_buildState->m_animationState != LegoVehicleBuildState::e_unknown4) {
+				if (m_unk0x108 > 0) {
+					DeleteObjects(&m_atomId, 500, 510);
+				}
+
+				m_unk0x258->SetUnknown0xbc(0);
+
+				if (GameState()->GetCurrentAct() == LegoGameState::e_act2) {
+					FUN_100243a0();
+				}
+				else if (m_unk0x258->AllPartsPlaced() || m_buildState->m_unk0x4d) {
+					m_buildState->m_unk0x4d = TRUE;
+					InvokeAction(Extra::e_start, m_atomId, m_carId, NULL);
+
+					{
+						// This pattern is suggested by BETA10 and also improves the LEGO1 match
+						MxNotificationParam param;
```

Comment:
Can param be directly instantiated within the function call instead of wrapping this in a scope, without worsening the match? This may not compile on modern compilers (we'd have to ifdef around that keeping this version for them) but seems more likely that they did that (have done so in other places)

Reply:
Great suggestion, this got the function to 100% - nice!